src/OrleansServiceBus/Providers/Streams/EventHub/EventHubPartitionCheckpointEntity.cs

```diff
@@ -0,0 +1,36 @@
+﻿
+using System;
+using Microsoft.ServiceBus.Messaging;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Orleans.ServiceBus.Providers
+{
+    internal class EventHubPartitionCheckpointEntity : TableEntity
+    {
+        public string Offset { get; set; }
+
+        public EventHubPartitionCheckpointEntity()
+        {
+            Offset = EventHubConsumerGroup.StartOfStream;
+        }
+
+        public static EventHubPartitionCheckpointEntity Create(string streamProviderName, string deploymentId, string partition)
+        {
+            return new EventHubPartitionCheckpointEntity
+            {
+                PartitionKey = MakePartitionKey(streamProviderName, deploymentId),
+                RowKey = MakeRowKey(partition)
+            };
+        }
+
+        public static string MakePartitionKey(string streamProviderName, string deploymentId)
+        {
+            return String.Format("EHCheckpoints_{0}_{1}", streamProviderName, deploymentId);
```

Comment:
is it worth sanitizing these (`AzureStorageUtils.SanitizeTableProperty`) either here or in the configuration?